File: community/sample-apps/golang-backend-server/event/event.go
package event

import (
	"context"
	"encoding/json"
	"fmt"
	"net/http"
	"time"

	"github.com/Khan/genqlient/graphql"

	"github.com/hasura/graphql-engine/community/sample-apps/golang-backend-server/generated"
)

type EventTriggerPayload[Old interface{}, New interface{}] struct {
	Event struct {
		SessionVariables struct {
			XHasuraRole string `json:"x-hasura-role"`
		} `json:"session_variables"`
		Op   string `json:"op"`
		Data struct {
			Old *Old `json:"old"`
			New *New `json:"new"`
		} `json:"data"`
		TraceContext struct {
			TraceID string `json:"trace_id"`
			SpanID  string `json:"span_id"`
		} `json:"trace_context"`
	} `json:"event"`
	CreatedAt    time.Time `json:"created_at"`
	ID           string    `json:"id"`
	DeliveryInfo struct {
		MaxRetries   int `json:"max_retries"`
		CurrentRetry int `json:"current_retry"`
	} `json:"delivery_info"`
	Trigger struct {
		Name string `json:"name"`
	} `json:"trigger"`
	Table struct {
		Schema string `json:"schema"`
		Name   string `json:"name"`
	} `json:"table"`
}

func NewUserHandler(w http.ResponseWriter, r *http.Request) {
	var u EventTriggerPayload[interface{}, struct {
		Id   string
		Name string
	}]
	err := json.NewDecoder(r.Body).Decode(&u)
	if err != nil {
		http.Error(w, err.Error(), http.StatusBadRequest)
		return
	}
	fmt.Println("Hello", u.Event.Data.New.Name)

	ctx := context.Background()
	client := graphql.NewClient("http://graphql-engine:8080/v1/graphql", http.DefaultClient)
	resp, _ := generated.GetUsers(ctx, client)
	for _, value := range resp.GetUser() {
		fmt.Printf("%#v", value)
	}

	w.WriteHeader(200)
}

File: community/sample-apps/python-backend-server/event/event.py
from pydantic import BaseModel, Field
from pydantic.generics import GenericModel
from typing import Generic, Literal, TypeVar

New = TypeVar("New", bound=BaseModel | None)
Old = TypeVar("Old", bound=BaseModel | None)


class DeliveryInfo(BaseModel):
    current_retry: int
    max_retries: int


class Data(GenericModel, Generic[New, Old]):
    new: New
    old: Old


class TraceContext(BaseModel):
    span_id: str
    trace_id: str


class Event(GenericModel, Generic[New, Old]):
    data: Data[New, Old]
    op: Literal["INSERT", "UPDATE", "DELETE", "MANUAL"]
    session_variables: dict[str, str]
    trace_context: TraceContext


class Table(BaseModel):
    name: str
    schema_: str = Field("", alias="schema")


class Trigger(BaseModel):
    name: str


class Payload(GenericModel, Generic[New, Old]):
    created_at: str
    delivery_info: DeliveryInfo
    event: Event[New, Old]
    id: str
    table: Table
    trigger: Trigger
